controllers/ibmpowervsmachinetemplate_controller_test.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2/klogr"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/cluster-api/util"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	infrav1beta1 "sigs.k8s.io/cluster-api-provider-ibmcloud/api/v1beta1"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestIBMPowerVSMachineTemplateReconciler_Reconcile(t *testing.T) {
+	testCases := []struct {
+		name                   string
+		expectError            bool
+		powerVSMachineTemplate *infrav1beta1.IBMPowerVSMachineTemplate
+		expectedCapacity       corev1.ResourceList
+	}{
+		{
+			name:        "Should Reconcile successfully if no IBMPowerVSMachineTemplate found",
+			expectError: false,
+		},
+		{
+			name:                   "Should Reconcile with memory and fractional processor values",
```

Comment:
Not sure whats wrong here these two tests cases are randomly flaky. Any idea?

Reply:
I could able to reproduce this issue in my local machine, I believe this is happening because patching is taking some time to reflect the actual status. This can be avoided by using `Eventually` block to test the status.